digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/service/RoadLinkService.scala

```diff
@@ -925,21 +925,21 @@ class RoadLinkService(val vvhClient: VVHClient, val eventbus: DigiroadEventBus,
       Some(roadLinks.minBy(roadlink => minimumDistance(point, roadlink.geometry)))
   }
 
-  def getClosestRoadlinkForCarTrafficFromVVH(user: User, point: Point): Seq[VVHRoadlink] = {
+  def getClosestRoadlinkForCarTrafficFromVVH(user: User, point: Point, forCarTraffic: Boolean = true): Seq[RoadLink] = {
     val diagonal = Vector3d(10, 10, 0)
 
     val roadLinks = user.isOperator() match {
-        case true =>  getVVHRoadLinks(BoundingRectangle(point - diagonal, point + diagonal))
-        case false => getVVHRoadLinks(BoundingRectangle(point - diagonal, point + diagonal), user.configuration.authorizedMunicipalities)
-      }
+      case true =>  getVVHRoadLinks(BoundingRectangle(point - diagonal, point + diagonal))
+      case false => getVVHRoadLinks(BoundingRectangle(point - diagonal, point + diagonal), user.configuration.authorizedMunicipalities)
+    }
 
-    roadLinks.isEmpty match {
-      case true => Seq.empty[VVHRoadlink]
-      case false => roadLinks.filter(rl => GeometryUtils.minimumDistance(point, rl.geometry) <= 10.0).filter(_.featureClass != FeatureClass.CycleOrPedestrianPath)
+    val closestRoadLinks = roadLinks.isEmpty match {
+      case true => Seq.empty[RoadLink]
+      case false => enrichRoadLinksFromVVH(roadLinks.filter(rl => GeometryUtils.minimumDistance(point, rl.geometry) <= 10.0))
     }
+    if (forCarTraffic) closestRoadLinks.filter(_.linkType != CycleOrPedestrianPath ) else closestRoadLinks
```

Comment:
I think this filtering could be done where it's called, in this case in verifyData

Reply:
I think we should use the method above getClosestRoadlinkFromVVH and change the `val diagonal = Vector3d(500, 500, 0)` to receive a parameter with a default value 500. Since the idea here when putting the forCarTraffic argument to false is the same as using the getClosestRoadlinkFromVVH  method.